src/observers/infoMessageObserver.ts

```diff
@@ -71,4 +83,62 @@ export class InfoMessageObserver {
                 ));
         }
     }
+
+    private handleExtensionActivated() {
+        if (this._environmentController.userType === UserType.Unknow) {
+            this.showInfoMessage(
+                `Are you a Microsoft employee or public contributor? We need the information to provide better validation experience. You can change it in Settings -> Extensions -> Doc Validation -> User type.`,
+                new MessageAction(
+                    "Microsoft internal employee",
+                    undefined,
+                    undefined,
+                    () => {
+                        const extensionConfig: vscode.WorkspaceConfiguration = vscode.workspace.getConfiguration(EXTENSION_NAME, undefined);
+                        extensionConfig.update(USER_TYPE, UserType.InternalEmployee, true);
+                    }
+                ),
+                new MessageAction(
+                    "Public contributor",
+                    undefined,
+                    undefined,
+                    () => {
+                        const extensionConfig: vscode.WorkspaceConfiguration = vscode.workspace.getConfiguration(EXTENSION_NAME, undefined);
+                        extensionConfig.update(USER_TYPE, UserType.PublicContributor, true);
+                    }
+                ));
+        }
+    }
+
+    private handleCommandWithUnkownUserTypeTriggered() {
+        if (this._environmentController.userType === UserType.Unknow) {
+            this.showInfoMessage(
+                `The command you just tried to operate needs your user type provided, please choose first. You can change it in Settings -> Extensions -> Doc Validation -> User type.`,

```

Comment:
suggestion: you are still able to change it by the extension settings (Docs validation -> User type) after this selection